options.go

```diff
@@ -262,7 +264,51 @@ func AddrsFactory(factory config.AddrsFactory) Option {
 		if cfg.AddrsFactory != nil {
 			return fmt.Errorf("cannot specify multiple address factories")
 		}
-		cfg.AddrsFactory = factory
+		if factory == nil {
+			return fmt.Errorf("cannot specify a nil address factory")
+		}
+
+		var log = logging.Logger("addrs-factory")
+
+		cfg.AddrsFactory = func(in []ma.Multiaddr) []ma.Multiaddr {
+			// This wrapper does the following:
+			// - If the user returns a webtransport multiaddr that doesn't have
+			//   a certhash, and all our known addrs have a certhash, copy the
+			//   certhash over.
+			//   - If we only have a webtransport multiaddr without a certhash
+			//     (client passed a static tls config), don't copy.
+			//   - If we have both, log a warning and don't copy the certhash.
```

Comment:
I usually don't review whitespace, but this looks really messy ;)

Reply:
reworded.